### ISSUE TITLE
How to return a final val loss in trainer?

### ISSUE THREAD

**Issue by Data-drone** [PyTorchLightning/pytorch-lightning#1942]
#### What is your question?

Most optimisation packages ie Ray Tune / Hyperopt return the train loop to return a final accuracy for the optimiser to decide what to try next.

How do I do this with the `Trainer` module for Pytorch Lightning?

#### What's your environment?

 - OS: Linux
 - Packaging pip
 - Version 0.7.6

**Comment by rajarajanvakil** [comment]
I too have the same issue , I want to  return "val_loss" in validation step and "avg_val_loss" from validation_epoch_end

**Comment by rajarajanvakil** [comment]
https://github.com/PyTorchLightning/pytorch-lightning/issues/321

`def validation_end(self, outputs):
        avg_loss = torch.stack([x['batch_val_loss'] for x in outputs]).mean()
        avg_acc = torch.stack([x['batch_val_acc'] for x in outputs]).mean()

        return {
          'val_loss': avg_loss,
          'val_acc': avg_acc, 
          'progress_bar':{'val_loss': avg_loss, 'val_acc': avg_acc }}`

**Comment by Data-drone** [comment]
I managed to achieve using callbacks and loggers but it doesn't work for ddp backend if doing distributed training. Think I need to include a manual gather step in there for that

**Comment by williamFalcon** [comment]
let’s formally support this somehow so it doesn’t have to be hacker around :)
